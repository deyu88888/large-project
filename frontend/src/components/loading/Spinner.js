import { jsx as _jsx } from "react/jsx-runtime";
const Spinner = () => {
    return (_jsx("div", { className: "flex justify-center items-center h-full", children: _jsx("div", { className: "animate-spin rounded-full h-12 w-12 border-t-4 border-blue-500" }) }));
};
export default Spinner;
