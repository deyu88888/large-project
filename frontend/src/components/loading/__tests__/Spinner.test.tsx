import React from 'react';
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Spinner from '../Spinner';

describe('Spinner Component', () => {
  it('renders the spinner correctly', () => {
    const { container } = render(<Spinner />);
    
    const spinnerContainer = container.firstChild;
    expect(spinnerContainer).toBeTruthy();
    
    expect(spinnerContainer).toHaveClass('flex');
    expect(spinnerContainer).toHaveClass('justify-center');
    expect(spinnerContainer).toHaveClass('items-center');
    expect(spinnerContainer).toHaveClass('h-full');
    
    const spinnerElement = spinnerContainer?.firstChild;
    expect(spinnerElement).toBeTruthy();
    
    expect(spinnerElement).toHaveClass('animate-spin');
    expect(spinnerElement).toHaveClass('rounded-full');
    expect(spinnerElement).toHaveClass('h-12');
    expect(spinnerElement).toHaveClass('w-12');
    expect(spinnerElement).toHaveClass('border-t-4');
    expect(spinnerElement).toHaveClass('border-blue-500');
  });

  it('matches snapshot', () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild).toMatchSnapshot();
  });
});