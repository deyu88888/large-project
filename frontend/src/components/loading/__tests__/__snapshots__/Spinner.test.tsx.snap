// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner Component > matches snapshot 1`] = `
<div
  class="flex justify-center items-center h-full"
>
  <div
    class="animate-spin rounded-full h-12 w-12 border-t-4 border-blue-500"
  />
</div>
`;
